### ISSUE TITLE
`brace-style` & `object-curly-newline`: infinite loop on typescript interface

### ISSUE THREAD

**Issue by baphony** [eslint-stylistic/eslint-stylistic#459]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

ESlint version: 9.6.0 (with flat config format)
@stylistic/eslint-plugin version: 2.3.0
File extension: .ts

Code:
```jsx
export interface SomeInterface {
    hasMethod(): boolean;
}
```
It gives that error on the curly brackets:
> Unexpected line break after this opening brace.eslint[@stylistic/object-curly-newline](https://eslint.style/rules/ts/object-curly-newline)

If I format with ESLint, I get this format:
```jsx
export interface UnreadMessagesPresenter { hasUnreadMessages(): boolean; }
```
It gives that error on the curly brackets:
> Statement inside of curly braces should be on next line.eslint[@stylistic/brace-style](https://eslint.style/rules/ts/brace-style)

If I format with ESLint, I go back to the first format 🔁 

Config rules:
- '@stylistic/object-curly-newline': ['error', { 'multiline': true }],
- '@stylistic/brace-style': 'error',

### Reproduction

sfsd

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by seahindeniz** [comment]
I also didn't get why there are two rules doing the same thing 🤔

**Comment by 9romise** [comment]
For this case, have a try with this config?
```
'@stylistic/object-curly-newline': ['error', { 'multiline': true }],
'@stylistic/brace-style': ['error', '1tbs', { 'allowSingleLine': true }],
```

**Comment by 9romise** [comment]
I opened an [issue](https://github.com/eslint-stylistic/eslint-stylistic/issues/415#755) to discuss whether TS nodes should be treated as `object-like` or `block-like`. Feel free to share your opinions.
